tests/30rooms/11leaving.pl

```diff
@@ -0,0 +1,292 @@
+            })
+        })->then( sub {
+            $do_request_json_for->( $user_b,
+                method => "PUT",
+                uri => "/api/v1/rooms/$room_id/state/m.room.name",
+                content => { "name" => "N1. B's room name before A left", },
+            )->on_done( sub { pass "User B set the room name the first time" } )
+        })->then( sub {
+            $do_request_json_for->( $user_b,
+                method => "PUT",
+                uri => "/api/v1/rooms/$room_id/state/madeup.test.state",
+                content => { "body" => "S1. B's state before A left", },
+            )->on_done( sub { pass "User B set the state the first time" } )
+        })->then( sub {
+            $do_request_json_for->( $user_b,
+                method => "POST",
+                uri => "/api/v1/rooms/$room_id/send/m.room.message",
+                content => {
+                    "body" => "M1. B's message before A left",
+                    "msgtype" => "m.room.text",
+                },
+            )->on_done( sub { pass "User B sent their first message" } )
+        })->then( sub {
+            $do_request_json_for->( $user_b,
+                method => "POST",
+                uri => "/api/v1/rooms/$room_id/send/m.room.message",
+                content => {
+                    "body" => "M2. B's message before A left",
+                    "msgtype" => "m.room.text",
+                },
+            )->on_done( sub { pass "User B sent their second message" } )
+        })->then( sub {
+            $do_request_json_for->( $user_a,
+                method => "POST",
+                uri => "/api/v1/rooms/$room_id/leave",
+                content => {},
+            )->on_done(sub { pass "User A left the room" })
+        })->then( sub {
+            $do_request_json_for->( $user_b,
+                method => "POST",
+                uri => "/api/v1/rooms/$room_id/send/m.room.message",
+                content => {
+                    "body" => "M3. B's message after A left",
+                    "msgtype" => "m.room.text",
+                },
+            )->on_done( sub { pass "User B sent their third message" } )
+        })->then( sub {
+            $do_request_json_for->( $user_b,
+                method => "PUT",
+                uri => "/api/v1/rooms/$room_id/state/m.room.name",
+                content => { "name" => "N2. B's room name after A left", },
+            )->on_done( sub { pass "User B set the room name the second time" } )
+        })->then( sub {
+            $do_request_json_for->( $user_b,
+                method => "PUT",
+                uri => "/api/v1/rooms/$room_id/state/madeup.test.state",
+                content => { "body" => "S2. B's state after A left", },
+            )->on_done( sub { pass "User B set the state the second time" } )
+        })
+    };
+
+
+test "A departed room is still included in /initialSync (SPEC-216)",
+    requires => [qw( do_request_json )],
```

Comment:
@leonerd didn't want to pass the room id as a dependency since it wasn't used outside this file. https://github.com/matrix-org/sytest/pull/17#discussion-diff-40563341
I suggest arguing with him.